lib/matplotlib/widgets.py

```diff
@@ -1933,15 +1933,31 @@ class SpanSelector(_SelectorWidget):
     rectprops : dict, default: None
         Dictionary of `matplotlib.patches.Patch` properties.
 
+    marker_props : dict
+        Properties with which the interactive handles are drawn.
+
     onmove_callback : func(min, max), min/max are floats, default: None
         Called on mouse move while the span is being selected.
 
     span_stays : bool, default: False
         If True, the span stays visible after the mouse is released.
+        Deprecated, use interactive instead.
+
+    interactive : bool, default: False
+        Whether to draw a set of handles that allow interaction with the
+        widget after it is drawn.
 
     button : `.MouseButton` or list of `.MouseButton`
         The mouse buttons which activate the span selector.
 
+    maxdist : float, default: 10
+        Distance in pixels within which the interactive tool handles can be
+        activated.
+
+    drag_from_anywhere : bool, optional
+        If `True`, the widget can be moved by clicking anywhere within
+        its bounds.
```

Comment:
I agree that this is the most natural behaviour, however, I don't understand how it would not change the current default (`drag_from_anywhere` doesn't exist and clicking inside the selector does nothing), your suggestion means that the current default will change when `interactive` is True and now it will move the selector.  With`interactive=False`, `drag_from_anywhere` doesn't do anything and I don't see how having `drag_from_anywhere=None` can help here!

Reply:
Sorry what I meant is to have a line like this:
```python
drag_from_anywhere = interactive if drag_from_anywhere is None else drag_from_anywhere
```

> d not change the current default (drag_from_anywhere doesn't exist and clicking inside the selector does nothing),

I suppose this is a behavior change, but I was considering it more as a UI change than an API change and thus  perhaps escaping the woes of backwards incompat - though I'm not the expert on where that line falls.

Reply:
The line depends on how angry it is likely to make a user who was expecting the old behavior and got the new one! 

I'm with @anntzer that I have used the old behavior (you can not move, only re-select and clicking gives a "zero width" selection which removes the selector), but still optimistic that we can find a middle ground.

Reply:
I took the liberty of pushing a commit to add back the ability to dismiss the selection by clicking somewhere else in the plot.

Reply:
> ```python
> drag_from_anywhere = interactive if drag_from_anywhere is None else drag_from_anywhere
> ```

This is equivalent to have `drag_from_anywhere=True` by default because when `interactive=False`, `drag_from_anywhere` is not used and therefore this is a behaviour change: if the user uses `span_stays=True` which is now `interactive=True` and they expect not to move the span selector when clicking on it.